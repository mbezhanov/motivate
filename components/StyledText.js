import React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import Colors from '../constants/Colors';
import Sizes from '../constants/Sizes';

const strokedText = props => (
  <View style={{ ...styles.container, ...props.style }}>
    <View style={styles.textWrapper}>
      <Text style={[styles.outline, { fontSize: props.size }]}>{props.children}</Text>
      <Text style={[styles.text, { fontSize: props.size }]}>{props.children}</Text>
    </View>
  </View>
);

const styles = StyleSheet.create({
  container: {
    margin: Sizes.M,
    alignItems: 'center',
  },
  textWrapper: {
    position: 'relative',
  },
  text: {
    color: Colors.HIGHLIGHT,
    textAlign: 'center',
  },
  outline: {
    color: Colors.OUTLINE,
    position: 'absolute',
    top: 2,
    left: 2,
    textAlign: 'center',
  }
});

export default strokedText;
