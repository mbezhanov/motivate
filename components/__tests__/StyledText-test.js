import React from 'react';
import renderer from 'react-test-renderer';
import StyledText from '../StyledText';

describe('StyledText', () => {
  it('renders correctly', () => {
    const styledText = renderer.create(<StyledText>Test 123</StyledText>).toJSON();
    expect(styledText).toMatchSnapshot();
  });

  it('renders text with a custom font size', () => {
    const props = { size: 123 };
    const styledText = renderer.create(<StyledText {...props}>Test 123</StyledText>).toJSON();
    expect(styledText).toMatchSnapshot();
  });

  it('accepts custom styling', () => {
    const style = {
      marginHorizontal: 123,
      marginVertical: 456,
    };
    const props = { style };
    const styledText = renderer.create(<StyledText {...props}>Test 123</StyledText>).toJSON();
    expect(styledText).toMatchSnapshot();
  });
});
