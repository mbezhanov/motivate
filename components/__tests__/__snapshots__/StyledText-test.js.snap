// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledText accepts custom styling 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "margin": 16,
      "marginHorizontal": 123,
      "marginVertical": 456,
    }
  }
>
  <View
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "left": 2,
            "position": "absolute",
            "textAlign": "center",
            "top": 2,
          },
          Object {
            "fontSize": 14,
          },
        ]
      }
    >
      Test 123
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#fff",
            "textAlign": "center",
          },
          Object {
            "fontSize": 14,
          },
        ]
      }
    >
      Test 123
    </Text>
  </View>
</View>
`;

exports[`StyledText renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "margin": 16,
    }
  }
>
  <View
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "left": 2,
            "position": "absolute",
            "textAlign": "center",
            "top": 2,
          },
          Object {
            "fontSize": 14,
          },
        ]
      }
    >
      Test 123
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#fff",
            "textAlign": "center",
          },
          Object {
            "fontSize": 14,
          },
        ]
      }
    >
      Test 123
    </Text>
  </View>
</View>
`;

exports[`StyledText renders text with a custom font size 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "margin": 16,
    }
  }
>
  <View
    style={
      Object {
        "position": "relative",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "left": 2,
            "position": "absolute",
            "textAlign": "center",
            "top": 2,
          },
          Object {
            "fontSize": 123,
          },
        ]
      }
    >
      Test 123
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#fff",
            "textAlign": "center",
          },
          Object {
            "fontSize": 123,
          },
        ]
      }
    >
      Test 123
    </Text>
  </View>
</View>
`;
